eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/BinarySpec.hs

```
@@ -69,3 +107,71 @@ spec = do
       prop (show (typeRep $ Proxy @t)) $ property $ \(x :: t) ->
         forM_ [eraProtVerLow @ConwayEra .. eraProtVerHigh @ConwayEra] $ \v ->
           decoderEquivalenceExpectation @t v (serialize v x)
+
+-- | Test empty MultiAsset rejection across protocol versions
+testMultiAssetRejection ::
+  forall era a.
+  ( ConwayEraImp era
+  , DecCBOR a
+  ) =>
+  Version -> -- Protocol version when rejection starts
+  String -> -- Era name for error messages
+  (Version -> BSL.ByteString) -> -- CBOR constructor function
+  Expectation
+testMultiAssetRejection rejectionVersion era mkCborBytes =
+  forM_ [eraProtVerLow @era .. eraProtVerHigh @era] $ \version -> do
+    let cborBytes = mkCborBytes version
+    case decodeFull @a version cborBytes of
+      Left _ ->
```

It is better to report this error
```suggestion
      Left err ->
```